libass/ass_cache.c

```diff
@@ -408,8 +448,29 @@ void *ass_cache_get(Cache *cache, void *key, void *priv)
     item->queue_next = NULL;
     item->ref_count = 2;
 
-    cache->cache_size += item->size;
     cache->items++;
+
+#if ENABLE_THREADS
+    assi_mutex_unlock(&cache->mutex);
+#endif
+
+    void *value = (char *) item + CACHE_ITEM_SIZE;
+    size_t size = desc->construct_func(new_key, value, priv);
+    assert(size);
+
+#if ENABLE_THREADS
+    assi_mutex_lock(&cache->mutex);
```

Comment:
I had to re-add this, because otherwise there was a race here:
```
            while (!item->size)
                pthread_cond_wait(&cache->cond, &cache->mutex);
```
where we could perform the `!item->size` check before the size was set, but then enter the wait after the cond had already been signaled.